dace/runtime/include/dace/xilinx/math.h

```diff
@@ -0,0 +1,11 @@
+/**
+    Additonal math function support for Xilinx
+*/
+
+#pragma once
+
+// Make abs work on Xilinx with vector types and compatible with Intel
+template <typename T>
+DACE_CONSTEXPR DACE_HDFI T fabs(const T& a) {
+    return (a < 0) ? a * -1 : a;
```

Comment:
`std::abs` returned garbage for vectors in my code, that is why I made the change in the first place.

Reply:
I see, but this doesn't work for vectors either. I'm not sure what `(a < 0)` even does in this case, but it certainly wouldn't be correct to negate the whole vector :-)

Reply:
Aren't this operators overloaded to operate element-wise?

Reply:
Even if all operations are elementwise, the implementation is not correct, because it's multiplying _all of_ `a` with `-1`, not just the elements that satisfied the condition :-)
If we want this to work for FPGA vectors, we should make an explicitly unrolled loop to do the element-wise abs

Reply:
Yeah I see that, could also overload the whole ternary to operate element-wise. That was the idea.

Reply:
You cannot overload the ternary operators in C++

Reply:
I suggest doing something like:
```cpp
template <typename T, int vector_length>
DACE_CONSTEXPR DACE_HDFI T fabs(const hlslib::DataPack<T, vector_length>& a) {
    hlslib::DataPack<T, vector_length> res;
    for (int i = 0; i < vector_length; ++i) {
        #pragma HLS UNROLL
        const auto elem = a[i];
        res[i] = elem < 0 ? -elem : elem;
    }
    return res;
}
```

Reply:
I don't see why this would not work, but it fails. I tested it. It produces wrong results. The unrolling somehow corrupts the memory or I don't know what is going on.

Reply:
I can see a bug in my suggestion above, so if you copied that it will be wrong :-) The return type should be `hlslib::DataPack<T, vector_length>`, not `T`

Reply:
Of course yes, resolved now.